File: src/main/java/com/jackmeng/stl/stl_ConsumableList.java
// Copyright 2023 Jack Meng. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package com.jackmeng.stl;

import java.util.ArrayList;

public class stl_ConsumableList< T > extends ArrayList< T >
{

  @Override public T get(int index)
  {
    T e = super.get(index);
    remove(index);
    return e;
  }
}

File: src/main/java/com/jackmeng/stl/stl_ConsumableList.kt
// Copyright 2023 Jack Meng. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.
package com.jackmeng.stl

import java.util.ArrayList

class stl_ConsumableList<T>:ArrayList<T>()
{
	override fun get(index:Int):T
	{
		val e=super.get(index)
		removeAt(index)
		return e
	}
}